README.md

```diff
@@ -1,96 +1,354 @@
-    },
-    "status": "SKIPPED",
-    "type": "DOCUMENT_STATUS"
+  test: () => {
+    // return true or false
   },
-  {
-    "data": [
-      {
-        "location": "example.tradetrust.io",
-        "status": "VALID",
-        "value": "0xe59877ac86c0310e9ddaeb627f42fdee5f793fbe"
-      }
-    ],
-    "name": "OpenAttestationDnsTxtIdentityProof",
-    "status": "VALID",
-    "type": "ISSUER_IDENTITY"
+  verify: async (document) => {
+    // perform checks and returns a fragment
   },
-  {
-    "name": "OpenAttestationDnsDidIdentityProof",
-    "reason": {
-      "code": 0,
-      "codeString": "SKIPPED",
-      "message": "Document was not issued using DNS-DID"
-    },
-    "status": "SKIPPED",
-    "type": "ISSUER_IDENTITY"
+};
+
+// create your own verify function with all verifiers and your custom one
+const verify = verificationBuilder([...openAttestationVerifiers, customVerifier], { network: "ropsten" });
+```
+
+Refer to [Extending Custom Verification](#extending-custom-verification) to find out more on how to create your own custom verifier.
+
+### Custom validation
+
+The `isValid` function will execute over fragments and determine if the fragments produced a valid result. By default the function will return true if a document fulfill the following conditions:
+
+- The document has NOT been tampered, AND
+- The document has been issued, AND
+- The document has NOT been revoked, AND
+- The issuer identity is valid.
+
+The function also allows a list of types to check for as a second parameter.
+
+```ts
+// index.ts
+import { isValid, openAttestationVerifiers, verificationBuilder } from "@govtechsg/oa-verify";
+import * as document from "./document.json";
+
+const verify = verificationBuilder(openAttestationVerifiers, {
+  network: "mainnet",
+});
+
+const fragments = await verify(document as any);
+
+console.log(isValid(fragments, ["DOCUMENT_INTEGRITY"])); // output true
+console.log(isValid(fragments, ["DOCUMENT_STATUS"])); // output false
+console.log(isValid(fragments, ["ISSUER_IDENTITY"])); // outpute false
+console.log(isValid(fragments)); // output false
+```
+
+- `isValid(fragments, ["DOCUMENT_INTEGRITY"])` returns true because the integrity of the document is not dependent on the network it has been published to.
+- `isValid(fragments, ["DOCUMENT_STATUS"])` returns false because the document has not been published on Ethereum main network.
+- `isValid(fragments, ["ISSUER_IDENTITY"])` returns false because there is no [DNS-TXT record](https://www.openattestation.com/docs/integrator-section/verifiable-document/ethereum/dns-proof) associated with the Ethereum main network's document store.
+- `isValid(fragments)` returns false because at least one of the above returns false.
+
+### Listening to individual verification method
+
+The `verify` function has an option to listen to individual verification methods. It might be useful if you want, for instance, to provide individual loader on your UI.
```

Comment:
```suggestion
The `verify` function provides an option to listen to individual verification methods. It might be useful if you want, for instance, to provide individual loaders on your UI.
```